Add Gitlab/Github feedback link icon in Techdocs Reader (#5159)
backstage/backstage: .changeset/techdocs-eight-camels-hear.md

---
'@backstage/plugin-techdocs': minor
---

> Add feedback link icon in Techdocs Reader that directs to Gitlab or Github repo issue page with prefilled title and source link.

Author: I haven't checked, i'll take a look at Bitbucket's system.

> Add feedback link icon in Techdocs Reader that directs to Gitlab or Github repo issue page with prefilled title and source link.

Author: I don't seem to be able to find the same functionality for Bitbucket i.e. prefill Bitbucket issue form by URL. 

I suggest this be left to another contributor with access to a Bitbucket instance to implement, as I don't have access to one at this time.

---
'@backstage/plugin-techdocs': minor
---

Add feedback link icon in Techdocs Reader that directs to Gitlab or Github repo issue page with prefilled title and source link.

> For link to appear, requires repo_url and edit_uri to be filled in mkdocs.yml, as per https://www.mkdocs.org/user-guide/configuration.

Reviewer: edit_uri does not seem to be required for GitHub, Bitbucket, and GitLab (at least for master branch for GitHub and GitLab, and default branch for Bitbucket) as per the note in https://www.mkdocs.org/user-guide/configuration/#edit_uri

Author: edit_uri apparently needs repo_url to have the common hostnames in order to be set automatically (eg gitlab.com, github.com). Self-hosting with other hostnames need a edit_uri specified.

Author: Added note on above.

Gitlab or Github detection checks for 'gitlab' or 'github' in hostname of source in repo_url;

> additionally 'techdocs.feedback.gitlab' and '.github' in app config can be used to specify hostnames that should be recognised as the respective Git hosting for the feedback link.

Reviewer 2: Should be updated to something like...

```suggestion
additionally, the 'host' key on 'gitlab' and 'github' configurations in the app 'integrations' config will be used to determine the feedback link.
```
